mapie/calibration.py

```diff
@@ -0,0 +1,502 @@
+                top_class_prob_,
+                y_calib_
+            )
+        else:
+            sample_weight_ = sample_weight
+        calibrator_ = fit_estimator(
+            calibrator_, top_class_prob_, y_calib_, sample_weight_
+        )
+        return calibrator_
+
+    def _get_calibrator(
+        self,
+        cv: Union[str, BaseCrossValidator],
+        calibrator: Optional[Union[str, RegressorMixin]],
+    ) -> RegressorMixin:
+        """
+        Check the input that has been provided for
+        calibrator and checks that the calibrator is a correct
+        estimator to calibrate.
+
+        Parameters
+        ----------
+        cv : Union[str, BaseCrossValidator]
+            Cross validation parameter.
+        calibrator : Union[str, RegressorMixin]
+            Calibrator as string to then be linked to one of the
+            valid methods otherwise calibrator as estimator.
+
+        Returns
+        -------
+        RegressorMixin
+            RegressorMixin to be used as calibrator.
+        Raises
+        ------
+        ValueError
+            If str is not one of the valid estimators.
+        """
+        if calibrator is None:
+            calibrator = "sigmoid"
+
+        if isinstance(calibrator, str):
+            if calibrator in self.valid_calibrators.keys():
+                calibrator = self.valid_calibrators[calibrator]
+            else:
+                raise ValueError(
+                    "Please provide a valid string from the valid calibrators."
+                )
+        check_estimator_fit_predict(calibrator)
+        return calibrator
+
+    def _get_labels(
+        self,
+        method: str,
+        pred: NDArray
+    ) -> Tuple[NDArray, NDArray]:
+        """
+        The "labels" is the way to create the different arrays needed
+        for the type of calibration method you want to make.
+
+        - Top-Label method means that you take the maximum probability
+        and calibrated each class when it is the maximum separately.
+
+        Parameters
+        ----------
+        method : str
+            The method that will be taken into account.
+        pred : NDArray of shape (n_samples, n_classes)
+            The output from the predict_proba of the classifier.
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray] of shapes (n_samples,) and (n_samples,)
+            The first element corresponds to the values that have to be
+            calibrated and the second element corresponds to class to be
+            associated.
+
+            In the Top-Label setting, the latter refers to the different
+            classes that have to be individually calibrated.
+
+        Raises
+        ------
+        ValueError
+            If the method provided has not been implemented.
+        """
+        if method == "top_label":
+            max_class_prob = np.max(pred, axis=1).reshape(-1, 1)
+            max_class_prob_arg = self.classes_[np.argmax(pred, axis=1)]
```

Comment:
why don't you take the result of the predict method of the model ?

Reply:
I think it would be better to just pass X, then as your estimator is in your self, you can call model.predict in this function without adding any other argument to your funciton

Reply:
done.